test/test/alloc/AllocTests.java

```diff
@@ -30,11 +30,9 @@ public void allocTotal(TestProcess p) throws Exception {
         Output out = p.profile("-e alloc -d 3 -o collapsed --total");
         assert out.samples("java.util.HashMap\\$Node\\[]") > 1_000_000;
 
-        /* Commented out as part of disabling asserting JFR output.
         out = p.profile("stop -o flamegraph --total");
-        assert out.contains("f\\(\\d+,\\d+,\\d+,\\d,'java.lang.Long'\\)");
-        assert out.contains("f\\(\\d+,\\d+,\\d+,\\d,'java.util.HashMap\\$Node\\[]'\\)");
-        */
+        assert out.contains("java\\.lang\\.Long");
```

Comment:
Implicit conversion of a flamegraph to collapsed output not only complicates the code, but also harms testing capabilities. For example, if I remove `-o flamegraph` from the above command, the test will still pass.
Tests should be explicit about output they expect. I propose adding `convert` method in the `Output` class which a test will explicitly call before calling `contains` etc.